railgun/manager/manager.go

```diff
@@ -0,0 +1,200 @@
+		return err
+	}
+	if err = (&kongctrl.KongConsumerReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("controllers").WithName("KongConsumer"),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "KongConsumer")
+		return err
+	}
+	*/
+
+	kongClient, err := kong.NewClient(&c.KongURL, http.DefaultClient)
+	if err != nil {
+		setupLog.Error(err, "unable to create kongClient")
+		return err
+	}
+
+	if err = (&kongctrl.SecretReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("controllers").WithName("Secret"),
+		Scheme: mgr.GetScheme(),
+		Params: kongctrl.SecretReconcilerParams{
+			WatchName:      c.SecretName,
+			WatchNamespace: c.SecretNamespace,
+			KongConfig: sendconfig.Kong{
+				URL:         c.KongURL,
+				FilterTags:  []string{c.FilterTag},
+				Concurrency: c.Concurrency,
+				Client:      kongClient,
+			},
+		},
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "Secret")
+		return err
+	}
+
+	// TODO - we've got a couple places in here and below where we "short circuit" controllers if the relevant API isn't available.
+	// This is convenient for testing, but maintainers should reconsider this before we release KIC 2.0.
+	// SEE: https://github.com/Kong/kubernetes-ingress-controller/issues/1101
+	if err := kongctrl.SetupIngressControllers(mgr); err != nil {
+		setupLog.Error(err, "unable to create controllers", "controllers", "Ingress")
+		return err
+	}
+
+	// TODO - similar to above, we're short circuiting here. It's convenient, but let's discuss if this is what we want ultimately.
+	// SEE: https://github.com/Kong/kubernetes-ingress-controller/issues/1101
+	udpIngressAvailable, err := kongctrl.IsAPIAvailable(mgr, &v1alpha1.UDPIngress{})
+	if !udpIngressAvailable {
+		setupLog.Error(err, "API configuration.konghq.com/v1alpha1/UDPIngress is not available, skipping controller")
+	} else {
+		if err = (&kongctrl.KongV1Alpha1UDPIngressReconciler{
+			Client: mgr.GetClient(),
+			Log:    ctrl.Log.WithName("controllers").WithName("UDPIngress"),
+			Scheme: mgr.GetScheme(),
+		}).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "UDPIngress")
+			return err
+		}
+	}
+
+	tcpIngressAvailable, err := kongctrl.IsAPIAvailable(mgr, &configurationv1beta1.TCPIngress{})
+	if !tcpIngressAvailable {
+		setupLog.Error(err, "API configuration.konghq.com/v1alpha1/TCPIngress is not available, skipping controller")
+	} else {
+		if err = (&kongctrl.KongV1Beta1TCPIngressReconciler{
+			Client: mgr.GetClient(),
+			Log:    ctrl.Log.WithName("controllers").WithName("TCPIngress"),
+			Scheme: mgr.GetScheme(),
+		}).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "TCPIngress")
+			return err
+		}
+	}
+	//+kubebuilder:scaffold:builder
```

Comment:
As with above, I expect this build tag will work just fine in this new location, but let's just double tap in case there's any project configuration changes needed to support this.

Reply:
We talked about this in Zoom this morning, given time constraints and a desire to focus on testing concerns for KIC 2.0, we're going to opt to intentionally sever controller management with kubebuilder for 2.0, instead using it only for API and config management for this release. We can re-connect the controller management later when we make more progress on splitting the ctrl monolith into separate microservices in later iterations.